3.2/docker-entrypoint.sh

```diff
@@ -51,9 +54,17 @@ case "$1" in
 				file_env 'REDMINE_DB_PASSWORD' "${POSTGRES_ENV_POSTGRES_PASSWORD}"
 				file_env 'REDMINE_DB_DATABASE' "${POSTGRES_ENV_POSTGRES_DB:-${REDMINE_DB_USERNAME:-}}"
 				file_env 'REDMINE_DB_ENCODING' 'utf8'
+			elif [ "$REDMINE_DB_SQLSERVER" ]; then
+				adapter='sqlserver'
+				host="$REDMINE_DB_SQLSERVER"
+				file_env 'REDMINE_DB_PORT' '1433'
+				file_env 'REDMINE_DB_USERNAME' "${POSTGRES_ENV_SQLSERVER_USER:-administrator}"
+				file_env 'REDMINE_DB_PASSWORD' "${POSTGRES_ENV_SQLSERVER_PASSWORD}"
+				file_env 'REDMINE_DB_DATABASE' "${POSTGRES_ENV_SQLSERVER_DB:-${REDMINE_DB_USERNAME:-}}"
```

Comment:
Same deal here as above -- these should all simply default to `''` since we don't need to support pulling credentials from a linked container automatically.

Reply:
I'm keen to understand this one better. How is this different from the MySQL or PostgreSQL case? If I go with `''`, will it still be possible to pass in these values when creating a Docker container based on this image?

FWIW, I myself am planning on using an external physical SQL server database, but I can see users wanting to be have the option of linking with a SQL Server container in one shot without having to mess with the `database.yml` file manually.

Reply:
Everyone will pass them is as `REDMINE_DB_USERNAME`,  `REDMINE_DB_PASSWORD`, etc (or the respective `_FILE` version), and we won't have a fallback to the `SQLSERVER_ENV_*` variables that come from the environment variables of a container linked as `--link my-sqlserver:sqlserver`.

Reply:
Thanks for explaining. So that entire section is just to establish fallbacks/defaults. Right on, I've made everything in that section `''` per @tianon's feedback.